### DISCUSSION TITLE
Error initializing CFWheels

### DISCUSSION THREAD

**Discussion by atspear** [wheels-dev/wheels#1262] [General]
I have a server that has been running for a long time with no changes, and now out of nowhere I'm getting the error below.  I'm running Lucee 5.3.10 and CFWheels 2.3.  I have no clue where to start troubleshooting this and any help would be appreciated.

Lucee 5.3.10.120 Error (expression)
--
Message | Component [wheels.Plugins] has no accessible Member with name [$CLASS]
Stacktrace | The Error Occurred in/web/default/wwwroot/wheels/plugins/initialization.cfm: line 239237:238: public any function getPlugins() {239: return variables.$class.plugins;240: }241:called from /web/default/wwwroot/wheels/global/internal.cfm: line 1013called from /web/default/wwwroot/wheels/events/onapplicationstart.cfm: line 925

**Comment by adamcameron** [comment]
This sounds - to me - like a symptom of CFWheels's incorrect approach to restarting itself (from the `reload` URL param).

See https://github.com/cfwheels/cfwheels/issues/1259.

If you reload the app that way, other requests will try to execute whilst `onApplicationStart` is running, during which the application is not in a completely initialised state. With really bad timing this can leave the app in a completely unstable state which will persist until the app is restarted _properly_ (CFML server restart, or calling `applicationStop`, and then letting the app reload properly.

When calling `reload` one needs to make sure the app is not processing any other traffic, basically.

**Reply by atspear** [reply]
Thanks for the reply.  Something like this happened a long time ago and I only remember that changing out the caching settings and clear some of the caches in Lucee fixed the issue, but that didn't work this time.  I ended up commenting out the "$loadPlugins()" call in onapplicationstart.cfm in the wheels directory.  It baffling because the server hasn't been touched in so long but something obviously changed.  Luckily I'm not using plugins with CFWheels, so this fix is fine but I'd love to figure out the root cause.
